src/transformState.js

```diff
@@ -1,11 +1,32 @@
 'use strict';
 
+const REMOVE_PROPERTIES = 'removeProperties';
+const ACTION_ADD_PROPERTIES = 'addProperties';
+
 /**
  * @param {Object} state
- * @param {Object[]} actions
+ * @typedef {Object} Action
+ * @property {string} type
+ * @property {Object} extraData
+ * @property {strinmg[]} keysToRemove
+ * @param {Action[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const key of actions) {
+    switch (key.type) {
+      case ACTION_ADD_PROPERTIES:
+        Object.assign(state, key.extraData);
+        break;
+      case REMOVE_PROPERTIES:
+        key.keysToRemove.forEach(item => delete state[item]);
+        break;
+      default:
+        Object.keys(state).forEach(item => delete state[item]);
```

Comment:
you're missing `clear` case - default case is supposed to throw new Error

Reply:
Looking for feedback from You :)